controllers/reconcile.go

```diff
@@ -598,95 +599,59 @@ func populateVolumes(d *apps.Deployment, frontend *crd.Frontend, frontendEnviron
 type ObjectStoreBucket struct {
 	AccessKey *string
 	SecretKey *string
-	Name      string
+	Name      *string
 	Region    *string
 	Endpoint  *string
 	Port      *string
 	TLS       *bool
 }
 
-// ExtractBucketConfigFromSecret extracts ObjectStoreBucket configuration for a specific bucket name
-// from a given Kubernetes Secret. It searches for the bucket name either directly
-// in the secret's Data map under the "bucket" key, or within the
-// "clowder/bucket-names" annotation (if it's a comma-separated list).
-//
-// It returns the ObjectStoreBucket configuration and an error if the secret data is
-// invalid or incomplete for the specified bucket, or if the bucket name isn't found.
-func ExtractBucketConfigFromSecret(secrets []v1.Secret, targetBucketName string) (*ObjectStoreBucket, error) {
-	if len(secrets) == 0 {
-		return nil, fmt.Errorf("no secrets provided to search for bucket '%s'", targetBucketName)
+// ExtractBucketConfigFromSecretByName extracts ObjectStoreBucket configuration from secrets
+func ExtractBucketConfigFromEnv() (*ObjectStoreBucket, error) {
+	// Required environment variables
+	accessKeyID := os.Getenv("PUSHCACHE_AWS_ACCESS_KEY_ID")
+	if accessKeyID == "" {
+		return nil, fmt.Errorf("required environment variable PUSHCACHE_AWS_ACCESS_KEY_ID is not set")
 	}
 
-	clowderBucketNamesAnnotation := "clowder/bucket-names"
-
-	for _, secret := range secrets {
-		currentSecret := secret
-
-		// Check if essential credential keys are present in the current secret's Data
-		requiredCredentialKeys := []string{
-			"aws_access_key_id",
-			"aws_secret_access_key",
-		}
-
-		credentialsPresent := true
-		for _, key := range requiredCredentialKeys {
-			if _, ok := currentSecret.Data[key]; !ok {
-				credentialsPresent = false
-				break
-			}
-		}
-		if !credentialsPresent {
-			continue
-		}
-
-		bucketNameFoundInThisSecret := false
-
-		// Check for "bucket" key in secret.Data
-		if secretBucketData, ok := currentSecret.Data["bucket"]; ok {
-			if string(secretBucketData) == targetBucketName {
-				bucketNameFoundInThisSecret = true
-			}
-		}
-
-		// If not found in Data, check "clowder/bucket-names" annotation
-		if !bucketNameFoundInThisSecret {
-			if annoValue, ok := currentSecret.Annotations[clowderBucketNamesAnnotation]; ok {
-				annotatedBucketNames := strings.Split(annoValue, ",")
+	secretAccessKey := os.Getenv("PUSHCACHE_AWS_SECRET_ACCESS_KEY")
+	if secretAccessKey == "" {
+		return nil, fmt.Errorf("required environment variable PUSHCACHE_AWS_SECRET_ACCESS_KEY is not set")
```

Comment:
@tahmidefaz I don't think we want to halt/crash the FEO pod if these aren't set (at least initially). It's not _currently_ required for the FEO to run correct? Once everything is in place on our multiple envs and we have several frontends using the new push cache, then maybe we mark these as required prior to turning off the legacy code path?

Also we can specify these as "required: true" in the parameters themselves in `deploy.yml`

Reply:
@florkbr This won't really halt the FEO pods since this gets hit when the `pushCacheEnabled` is set to `true` in the CRD. If the `pushCacheEnabled` is `true` and these env vars aren't present, the frontend pods just won't get reconciled, but the FEO pods will still be up.

https://github.com/RedHatInsights/frontend-operator/blob/main/controllers/reconcile.go#L110